### PR TITLE
fix(api-client): lower syntax highlighting threshold on responseBody

### LINKED ISSUE #3467: Random/Intermitment UI freezing happens when displaying large JSON responses
### What happens?

**Version:** 1.25.34

Noticable random and long freezing When testing any API endpoint that returns large JSON responses (~ 24 KiB download size).
It costs at least 3 seconds to display a response with 46 ms (round trip time).
Then, every UI components seems freezing, even hovering states are not displaying correctly.
After a few extra seconds, I can start scrolling the response view, but it may or may not scroll properly.
Using the small scrollbar for quickly navigating is completely broken, it has been frozen.
The UI may be unfrozen after a while or not, even on a very beefy workstation with 64 GiB RAM and a lot of processing cores.

### What did you expect to happen?

I think we should disable syntax highlighting for large JSON responses (similar to VS Code behaviors when opening very big files).
I see this kind of issues had been addressed, but not sure why Scalar UI cannot handle ~ 24 KiB of JSON responses without freezing.

https://github.com/scalar/scalar/issues/2503
https://github.com/scalar/scalar/issues/2333

### OpenAPI Document

_No response_

### PR DESCRIPTION
**Pull request by amritk**
Fixes #3467

Switched the unit of measurement to bytes and dropped the threshold to 20k. Lets see how this does.

Also removed some Json parsing since it was redundant. Will fix up the tests

### CODE REVIEW

File: packages/api-client/src/libs/send-request.ts [hunk 1]
```diff
@@ -32,11 +32,6 @@ export function decodeBuffer(buffer: ArrayBuffer, contentType: string) {
     const decoder = new TextDecoder(mimeType.parameters.get('charset'))
     const string = decoder.decode(buffer)
 
-    // JSON
```
Comment (amritk) [inline packages/api-client/src/libs/send-request.ts:35]: We convert it later anyway, no need here

File: packages/api-client/src/libs/send-request.ts [hunk 2]
```diff
@@ -218,9 +213,9 @@ export function createFetchBody(
 }
 
 /** Response from sendRequest hoisted so we can use it as the return type for createRequestOperation */
-type SendRequestResponse<ResponseDataType = unknown> = Promise<
```
Comment (amritk) [inline packages/api-client/src/libs/send-request.ts:221]: removed this extra typing as its not needed

**Comment by hanspagel** [comment]
Awesome! Tried to fix the tests, let’s see. Unfortunately, the tests became uglier, because we need to JSON.parse everywhere now, but it might still be better to keep the ugly tests and remove it from the actual function.

**Review by hanspagel — APPROVED** [review]
Love everything that improves the performance

**Pull request merged**